sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -424,7 +424,7 @@ private[sql] object PartitioningUtils {
     path.foreach { c =>
       if (needsEscaping(c)) {
         builder.append('%')
-        builder.append(f"${c.asInstanceOf[Int]}%02x")
+        builder.append(f"${c.asInstanceOf[Int]}%02X")
```

Comment:
@liancheng So i was comparing the output of our impl against hive. And hive reports the escaped name in upper case,  I looked at FileUtils.escapePathName() in hive as reference before changing the code here. Please let me know what you think.